numba/cuda/codegen.py

```diff
@@ -232,7 +240,8 @@ def finalize(self):
         # https://github.com/numba/numba/pull/890
         for library in self._linking_libraries:
             for fn in library._module.functions:
-                if not fn.is_declaration:
+                if (not fn.is_declaration and
```

Comment:
comment we only adjust the linkage when non-debug, because the device functions need external linkage when linked separately